### PR TITLE
Post comment as notification if is deploy from a PR.

### PR DESCRIPTION
**Pull request by shrimpy**
Comments will be look like below screenshot, there will be two message depends on if deployment has continuous deployment enabled or not.

image

### CODE REVIEW

File: Slingshot.Api/Controllers/ARMController.cs [hunk 1]
```diff
@@ -395,6 +396,35 @@ private void DeleteResourceGroup(string subscriptionId, string token, string rgN
...
+        public async Task DeploymentNotification(DeployInputs inputs)
+        {
+            string repositoryUrl = HttpUtility.UrlDecode(inputs.repoUrl);
+            string token = GetTokenFromHeader();
+            Repository repo = Repository.CreateRepositoryObj(repositoryUrl, Request.RequestUri.Host, token);
+            var queryStrings = HttpUtility.ParseQueryString(repositoryUrl);
+            if (queryStrings["pr"] != null)
+            {
+                // if deployment is come from a pull request, post a comment back to the pull request.
+                string siteUrl = string.Format(CultureInfo.InvariantCulture, "https://{0}.azurewebsites.net", inputs.parameters["siteName"]["value"]);
```
Comment (davidebbo) [inline Slingshot.Api/Controllers/ARMController.cs]: Ideally, avoid hard coding azurewebsites.net and use same logic as what we already have to give user link to new site

File: Slingshot.Api/Controllers/ARMController.cs [hunk 2]
```diff
@@ -395,6 +396,35 @@ private void DeleteResourceGroup(string subscriptionId, string token, string rgN
...
+                bool isManualIntegration = true;
+                if (inputs.parameters["isManualIntegration"] != null &&
+                    inputs.parameters["isManualIntegration"]["value"] != null &&
+                    bool.TryParse(inputs.parameters["isManualIntegration"]["value"].ToString(), out isManualIntegration) &&
+                    !isManualIntegration)
+                {
+                    pullRequestComment.Append(" with continuous deployment enabled");
+                }
+
+                pullRequestComment.AppendFormat(". [{0}]({0})", siteUrl);
```
Comment (davidebbo) [inline Slingshot.Api/Controllers/ARMController.cs]: They already turn straight URLs into links, so you you don't need URL markdown syntax.

File: Slingshot.Api/Abstract/Repository.cs [hunk 3]
```diff
@@ -146,6 +146,11 @@ public virtual Task<bool> HasScmInfo()
             return Task.FromResult(false);
         }
 
+        public virtual Task WritePullRequestComment(string prId, string comment)
+        {
+            return Task.FromResult(new object());
```
Comment (ahmelsayed) [inline Slingshot.Api/Abstract/Repository.cs]: There is `Task.CompletedTask` in .NET 4.6. Not sure which framework this is targeting.

Comment (shrimpy) [inline Slingshot.Api/Abstract/Repository.cs]: currently targeting .net 4.5
might be a good idea to upgrade this project to 4.6

i will create a separate work item for it.

Comment (shrimpy) [inline Slingshot.Api/Abstract/Repository.cs]: here it is https://github.com/projectkudu/slingshot/issues/56

**Pull request merged**